src/cpp/shared_core/include/shared_core/Logger.hpp

```diff
@@ -374,6 +374,8 @@ void removeLogDestination(const std::string& in_destinationId, const std::string
  */
 void removeReloadableLogDestinations();
 
+bool needsDebugLog();
```

Comment:
How do you feel about something like `bool isAtLeastLogLevel(LogLevel in_level);` so it can be generalized for other use cases in the future?

Nit: This file is currently sorted alphabetically, could you move this to keep that ordering?

Reply:
I couldn't think of a use case for the other log levels so I put the focus on readability. I'm happy to make the change, though if you prefer.  How about:

  if (log::needsLogLevel(log::LogLevel::DEBUG))

It seems a little more readable to me.  Thanks for catching those ordering problems. I'll fix those up for sure!

Reply:
`needsLogLevel` doesn't feel quite right to me since the logger doesn't need a log level. We could make it shorter (e.g. `isLogLevel`). In pseudo code, I would mentally read them this way:

`if logger needs log level debug, then <blah>`
vs.
`if logger is log level debug, then <blah>`

The second one feels slightly more correct to me, but the actual name is really a nit, so up to you.

Reply:
I like your explanation so I changed it to isLogLevel(LogLevel::DEBUG) 

I also tried to fix the ordering and removed an unused declaration for "cleanDelims". Thanks!